src/components/UserSearch.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import {Route} from 'react-router-dom';
+
+class UserSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      githubUsername: '',
+      selectedUsername: '',
```

Comment:
Okay that makes sense. To avoid keeping up with callbacks that will go update other components, those components can just have access to that data through global state. I was wondering how the other component was going to know about the state change if it was local, I can take a crack at refactoring this.